### PR TITLE
refactor(api-gen): parse links with descriptions

### PR DESCRIPTION
**Pull request by JeanMeche**
We already parsed the links correctly for the pills but didn't for function descriptions

2nd commits also support markdown links with titles. 

fixes angular/angular#56856
fixes angular/angular#56857

### CODE REVIEW

File: bazel/api-gen/rendering/templates/docs-pill-row.tsx [hunk 1]
```diff
@@ -15,11 +15,11 @@ export function DocsPillRow(props: {links: LinkEntryRenderable[]}) {
 
   return (
     <nav class="docs-pill-row">
-      {
-        props.links.map((link) => (<a class="docs-pill" href={link.url}>
+      {props.links.map((link) => (
+        <a class="docs-pill" href={link.url} title={link.title}>
```
Comment (josephperrott) [inline bazel/api-gen/rendering/templates/docs-pill-row.tsx:19]: Are we going to run into issues if we have `title=` because `link.title` is undefined?

**Review by josephperrott — APPROVED** [review]
LGTM

File: bazel/api-gen/rendering/templates/docs-pill-row.tsx (hunk 1 above)
Comment (JeanMeche) [reply bazel/api-gen/rendering/templates/docs-pill-row.tsx:19]: afaik it won't be rendered if its `undefined`.

Comment (josephperrott) [reply bazel/api-gen/rendering/templates/docs-pill-row.tsx:19]: perfect, just wanted to sanity check.

**Pull request closed without merging**